### ISSUE TITLE
IPC broken in v3.3.1

### ISSUE THREAD

**Issue by neeasade** [qutebrowser/qutebrowser#8366]
**Version info**:

```
qutebrowser v3.3.1
Git commit: 
Backend: QtWebEngine 6.8
  based on Chromium 122.0.6261.171
  with security patches up to 129.0.6668.70 (plus any distribution patches)
  (source: api)
```

paste: https://bpa.st/TZZO4

**Does the bug happen if you start with `--temp-basedir`?**: yes

**Description**
Send a command to the ipc socket and see that nothing happens (`${XDG_RUNTIME_DIR}/qutebrowser/ipc-$(echo -n "$USER" | md5sum | cut -d' ' -f1)`)

I rolled back to confirm it is working in v3.2.1, `:version` pastes:

working: https://bpa.st/Z7QF6
broken: https://bpa.st/TZZO4

**How to reproduce**
Test with the script `open_url_in_instance.sh`: https://github.com/qutebrowser/qutebrowser/blob/main/scripts/open_url_in_instance.sh , see that nothing happens

**Comment by The-Compiler** [comment]
Works fine for me. A couple of questions:

- How did you test this with `--temp-basedir`? Did you just run the script normally? That's expected to not work FWIW, as the basedir is part of the socket name.
- Can you try keeping the Qt version at 6.7/6.8 and only changing the qutebrowser version, or vice-versa? Nothing big changed about IPC in qutebrowser for a long time.
- Can you start qutebrowser with `--debug --logfilter ipc` and show the log when you run the script?
- Does IPC work if you run `qutebrowser https://example.org` rather than using that script? If not, can you also add `--debug --logfilter ipc` to that process and show both logs?
- What do those print?
    - `echo ${XDG_RUNTIME_DIR}/qutebrowser/ipc-$(echo -n "$USER" | md5sum | cut -d' ' -f1)`
    - `ls $XDG_RUNTIME_DIR/qutebrowser`

**Comment by The-Compiler** [comment]
@neeasade any updates?

**Comment by neeasade** [comment]
oh jeez. I was taking a look at this again this morning (thank you for all the details to investigate!!) and the issue turned out to be an exit condition in my script :weary:. 

TLDR I had an exit condition based on `pgrep qutebrowser` but qutebrowser was being run as `python3 /nix/store/..../bin/qutebrowser.wrapped`

Thanks again for the pointers, I use the IPC for reloading themes and abusing the adblock for a "focus mode" (block social media temporarily)

**Comment by The-Compiler** [comment]
Hah, thanks for the detailed update in any case!
